quinn-proto/src/connection/streams/state.rs

```diff
@@ -536,7 +537,7 @@ impl StreamsState {
                 // If the stream still has pending data, reinsert it, possibly with an updated priority value
                 // Fairness with other streams is achieved by implementing round-robin scheduling,
                 // so that the other streams will have a chance to write data before we touch this stream again.
-                self.pending.push_pending(id, stream.priority);
+                self.pending.reinsert_pending(id, stream.priority);
```

Comment:
nit: pass `fair` down here to save the extra field in `PendingStreamsQueue`?

Reply:
good idea, done